### PR TITLE
create CRDs while controller starts if they are not exist

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

create CRDs while controller starts if they are not exist

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @cd1989 @supereagle @caicloud/chenpengyu-write 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/cyclone/v1alpha1/crd.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	createCRD("stagetemplate", "stagetemplates", "StageTemplate", []string{"stpl"}, client)
+	createCRD("workflowparam", "workflowparams", "WorkflowParams", []string{"wfp"}, client)
+	createCRD("workflowtrigger", "workflowtriggers", "WorkflowTrigger", []string{"wft"}, client)
+}
+
+func createCRD(singular, plural, kind string, shortNames []string, client apiextensionsclient.Interface) {
+	crdName := plural + "." + GroupName
+	_, err := client.ApiextensionsV1beta1().CustomResourceDefinitions().Get(crdName, metav1.GetOptions{})
+	if err != nil {
+		if err.(*errors.StatusError).Status().Reason == metav1.StatusReasonNotFound {
```
Comment (cd1989) [inline pkg/apis/cyclone/v1alpha1/crd.go]: We can use `errors.IsNotFound` directly.

File: pkg/apis/cyclone/v1alpha1/crd.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+					},
+				},
+			})
+			if err != nil {
+				log.WithField("name", crdName).WithField("error", err).Fatal("create crd error")
+			}
+		} else {
+			log.WithField("name", crdName).WithField("error", err).Fatal("check existence of crd error")
+		}
+	} else {
```
Comment (cd1989) [inline pkg/apis/cyclone/v1alpha1/crd.go]: `else` after long statements block is not recommended.   I suggest,

```
if err == nil {
  log.WithField("name", crdName).Info("crd already exist")
  return
}

if !errors.IsNotFound(err) {
  log.WithField("name", crdName).WithField("error", err).Fatal("check existence of crd error")
}

... create crd here...
```

File: pkg/apis/cyclone/v1alpha1/crd.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		if err.(*errors.StatusError).Status().Reason == metav1.StatusReasonNotFound {
+			log.WithField("name", crdName).Info("create crd")
+			_, err := client.ApiextensionsV1beta1().CustomResourceDefinitions().Create(&v1beta1.CustomResourceDefinition{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "CustomResourceDefinition",
+					APIVersion: "apiextensions.k8s.io/v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      crdName,
+					Namespace: "default",
```
Comment (supereagle) [inline pkg/apis/cyclone/v1alpha1/crd.go]: Not all CRDs are namespace scope.

File: pkg/apis/cyclone/v1alpha1/crd.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+			return nil, err
+		}
+	}
+
+	return apiextensionsclient.NewForConfigOrDie(config), nil
+}
+
+// EnsureCRDCreated will create built-in CRDs if they are not exist.
+func EnsureCRDCreated(masterUrl, kubeConfigPath string) {
+	log.Info("start to creat crd")
```
Comment (supereagle) [inline pkg/apis/cyclone/v1alpha1/crd.go]: s/creat/create/

File: pkg/apis/cyclone/v1alpha1/crd.go (hunk 3 above)
Comment (zhujian7) [reply pkg/apis/cyclone/v1alpha1/crd.go]: Which CRD do you mean?

**Pull request merged**